terraform/environments/staging/demo.tf

```diff
@@ -0,0 +1,179 @@
+# This module deploys an empty VM ready to run Docker commands to deploy our Relay or Gateway,
+# it's used weekly for internal demos and testing, and reset after each use by rebooting the VM.
+
+data "google_compute_image" "demo" {
+  family  = "ubuntu-2204-lts"
+  project = "ubuntu-os-cloud"
+}
+
+resource "google_service_account" "demo" {
+  project = module.google-cloud-project.project.project_id
+
+  account_id   = "demo-instance"
+  display_name = "Custom Service Account for a Demo VM Instance"
+}
+
+resource "google_compute_instance" "demo" {
+  project = module.google-cloud-project.project.project_id
+
+  name         = "demo"
+  machine_type = "n1-standard-1"
+  zone         = "${local.region}-d"
+
+  tags = ["demo"]
+
+  boot_disk {
+    auto_delete = true
+
+    initialize_params {
+      image = data.google_compute_image.demo.self_link
+
+      labels = {
+        managed_by = "terraform"
+      }
+    }
+  }
+
+  network_interface {
+    subnetwork = google_compute_subnetwork.apps.self_link
+
+    access_config {
+      // Ephemeral public IP
+    }
+  }
+
+  metadata = {
+    google-logging-enabled       = "true"
+    google-logging-use-fluentbit = "true"
+
+    # Report health-related metrics to Cloud Monitoring
+    google-monitoring-enabled = "true"
+  }
+
+  # We can install any tools we need for the demo in the startup script
+  metadata_startup_script = <<EOT
+  set -xe \
+    && sudo apt update -y \
+    && sudo apt install postgresql-client jq iperf3 -y \
+    && sudo apt install apt-transport-https ca-certificates curl software-properties-common -y \
+    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg \
+       | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" \
+       | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null \
+    && sudo apt update -y \
+    && sudo apt install docker-ce docker-ce-cli containerd.io -y \
+    && sudo usermod -aG docker $(whoami) \
+    && sudo systemctl enable docker \
+    && sudo systemctl start docker \
+    && sudo docker run -d --restart always --name=httpbin -p 80:80 kennethreitz/httpbin \
+    && echo ${module.metabase.internal_ip} metabase.fz >> /etc/hosts
```

Comment:
This allows us to create a:
- DNS resource for `metabase.fz`
- CIDR resource for apps subnet `10.128.0.0/20`
- IP resource for Postgres database (see `terraform outputs`)
- localhost iperf3 resource `127.0.0.1`
- locahost docker container resource (HTTPBin) `localhost:80`